SomerenDAL/ActivityDao.cs

```diff
@@ -12,8 +12,7 @@ public List<Activity> GetAllActivities()
         {
             string query = "SELECT [activityId], [start_time], [end_time], [activity_name] FROM Activities";
             SqlParameter[] sqlParameters = new SqlParameter[0];
-            DataTable dataTable = ExecuteSelectQuery(query, sqlParameters);
-            return ReadTables(dataTable);
+            return ReadTables(ExecuteSelectQuery(query, sqlParameters));
```

Comment:
I would like to keep the two-line approach for the following reasons:
1. Debugging: It's easier to inspect the value of dataTable in a debugger when it's assigned to a variable on its own line.
2. If you need to add error handling of the dataTable before passing it to ReadTables, having it as a separate line makes this insertion simpler.
3. This can be slightly harder to read, especially for complex method calls or when chaining multiple methods together, as it packs more logic into a single line.